### PR TITLE
Refactor exit code and signal of ToolRunner

### PR DESCRIPTION
**Pull request by hyunsik-yoon**
Currently exit code and signal handling is kinda complex.
This commit tries to sumplify this by introducing two
interfaces.

from https://github.com/Samsung/ONE-vscode/pull/749#issuecomment-1149432151

ONE-vscode-DCO-1.0-Signed-off-by: Hyun Sik Yoon <hyunsik.yoon.1024@gmail.com>

### CODE REVIEW

**Review by jyoungyun — DISMISSED** [review]
LGTM

File: src/Project/ToolRunner.ts [hunk 1]
```diff
@@ -25,8 +25,29 @@ const which = require('which');
...
+  // When this process was intentionally killed by user, this must be true.
+  intentionallyKilled?: boolean;
+}
+
+/**
+ * Return type when a process exits with error
+ * One of exitCode or signal must NOT be undefined.
+ */
+export interface ErrorResult {
+  // When successful exit, exit code must be greater than 0
```
Comment (seanshpark) [inline src/Project/ToolRunner.ts:44]: (optional) `successful` with Error seems strange :) Maybe `error` ?

**Review by seanshpark — APPROVED** [review]
LGTM

File: src/Project/ToolRunner.ts [hunk 2]
```diff
@@ -25,8 +25,29 @@ const which = require('which');
...
+  exitCode?: number;
+  // When this process was killed by, e.g., kill command from shell,
+  // this must be set to proper NodeJS.Signals.
+  signal?: NodeJS.Signals;
+}
+;
```
Comment (YongseopKim) [inline src/Project/ToolRunner.ts:50]: a nit. `;`?

Comment (YongseopKim) [reply src/Project/ToolRunner.ts:50]: it can be ignored

Comment (YongseopKim) [reply src/Project/ToolRunner.ts:50]: (Can be ignored my review)

**Comment by hyunsik-yoon** [comment]
let me fix the following in the next PR
- @seanshpark's https://github.com/Samsung/ONE-vscode/pull/760#discussion_r892016350
- @YongseopKim's https://github.com/Samsung/ONE-vscode/pull/760#discussion_r892048833

Please merge. Thanks. :-)

File: src/Project/ToolRunner.ts (hunk 1 above)
Comment (hyunsik-yoon) [reply src/Project/ToolRunner.ts]: Fixed. Thanks. :-)

File: src/Project/ToolRunner.ts (hunk 2 above)
Comment (hyunsik-yoon) [reply src/Project/ToolRunner.ts]: Fixed. Thanks. :-)

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by YongseopKim — APPROVED** [review]
LGTM

**Pull request merged**